clients/src/test/java/org/apache/kafka/clients/MockClient.java

```diff
@@ -75,15 +75,7 @@ public FutureResponse(Node node,
     private int correlation;
     private final Time time;
     private final MockMetadataUpdater metadataUpdater;
-    private final Set<String> ready = new HashSet<>();
-
-    // Nodes awaiting reconnect backoff, will not be chosen by leastLoadedNode
-    private final TransientSet<Node> blackedOut;
-    // Nodes which will always fail to connect, but can be chosen by leastLoadedNode
-    private final TransientSet<Node> unreachable;
-    // Nodes which have a delay before ultimately succeeding to connect
-    private final TransientSet<Node> delayedReady;
-
+    private final Map<String, ConnectionState> connections = new HashMap<>();
```

Comment:
Just to clarify this refactoring is orthogonal to unit test augment for this bug fix specifically right?

Reply:
I needed it in order to add a good test case.

Reply:
ack.